IdSSLOpenSSL.pas

```diff
@@ -3299,7 +3284,24 @@ procedure TIdSSLContext.InitContext(CtxMode: TIdSSLCtxMode);
       SSL_CTX_clear_options(fContext, SSL_OP_NO_TLSv1_2);
     end;
   end;
+  if IsOpenSSL_TLSv1_3_Available then begin
+    if not(sslvTLSv1_3 in SSLVersions) then begin
+      SSL_CTX_set_options(fContext, SSL_OP_NO_TLSv1_3);
+    end 
+	else if (fMethod = sslvSSLv23) then begin
+      SSL_CTX_clear_options(fContext, SSL_OP_NO_TLSv1_3);
+    end;
+  end;
 
+  if sslvTLSv1_3 in SSLVersions then
+    SSL_CTX_set_min_proto_version(fContext, TLS1_2_VERSION)

```

Comment:
Typo? Should this be `TLS1_3_VERSION` instead?

Reply:
No, it's not a typo, I set TLS 1.2 as minimum version to work as a fallback for servers that doesn't support TLS 1.3. So first tries to use TLS 1.3, if fails, tries to connect using TLS 1.2. But you can change to TLS1_3_VERSION if you prefer, this was set some years ago when TLS 1.3 was almost not used.